test/unit/api_spec.js

```diff
@@ -1053,112 +937,89 @@ describe("api", function () {
         const page3Actions = await page3.getJSActions();
         return [docActions, page1Actions, page3Actions];
       });
-      promise
-        .then(async ([docActions, page1Actions, page3Actions]) => {
-          expect(docActions).toEqual({
-            DidPrint: [`this.getField("Text2").value = "DidPrint";`],
-            DidSave: [`this.getField("Text2").value = "DidSave";`],
-            WillClose: [`this.getField("Text1").value = "WillClose";`],
-            WillPrint: [`this.getField("Text1").value = "WillPrint";`],
-            WillSave: [`this.getField("Text1").value = "WillSave";`],
-          });
-          expect(page1Actions).toEqual({
-            PageOpen: [`this.getField("Text1").value = "PageOpen 1";`],
-            PageClose: [`this.getField("Text2").value = "PageClose 1";`],
-          });
-          expect(page3Actions).toEqual({
-            PageOpen: [`this.getField("Text5").value = "PageOpen 3";`],
-            PageClose: [`this.getField("Text6").value = "PageClose 3";`],
-          });
-          loadingTask.destroy().then(done);
-        })
-        .catch(done.fail);
+      const [docActions, page1Actions, page3Actions] = await promise;
```

Comment:
Nit: In *this* particular case, given how the pre-existing code is written above, I'm not sure that we actually need the temporary `promise` for anything?

*Normally* we usually look-up the various resources in *parallel*, but in this case everything is done serially and the following should thus be equivalent:
```javascript
const pdfDoc = await loadingTask.promise;
const docActions = await pdfDoc.getJSActions();
const page1 = await pdfDoc.getPage(1);
const page1Actions = await page1.getJSActions();
const page3 = await pdfDoc.getPage(3);
const page3Actions = await page3.getJSActions();
```